src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -217,28 +160,24 @@ internal abstract partial class AbstractNavigateToSearchService
             allPatternMatches.ToImmutableAndClear());
     }
 
-    private static async ValueTask<ImmutableArray<ProjectId>> GetAdditionalProjectsWithMatchAsync(
-        Document? document, DeclaredSymbolInfo declaredSymbolInfo, CancellationToken cancellationToken)
+    private static ImmutableArray<ProjectId> GetAdditionalProjectsWithMatch(
+        Document? document,
+        DeclaredSymbolInfo declaredSymbolInfo,
+        ArrayBuilder<(TopLevelSyntaxTreeIndex, ProjectId)>? linkedIndices)
     {
-        if (document == null)
+        if (document == null || linkedIndices is null || linkedIndices.Count == 0)
             return [];
 
-        using var _ = ArrayBuilder<ProjectId>.GetInstance(out var result);
+        using var result = TemporaryArray<ProjectId>.Empty;
 
-        var solution = document.Project.Solution;
-        var linkedDocumentIds = document.GetLinkedDocumentIds();
-        foreach (var linkedDocumentId in linkedDocumentIds)
+        foreach (var (index, projectId) in linkedIndices)
         {
-            var linkedDocument = solution.GetRequiredDocument(linkedDocumentId);
-            var index = await TopLevelSyntaxTreeIndex.GetRequiredIndexAsync(linkedDocument, cancellationToken).ConfigureAwait(false);
-
             // See if the index for the other file also contains this same info.  If so, merge the results so the
             // user only sees them as a single hit in the UI.
             if (index.DeclaredSymbolInfoSet.Contains(declaredSymbolInfo))
-                result.Add(linkedDocument.Project.Id);
+                result.Add(projectId);
         }
 
-        result.RemoveDuplicates();
         return result.ToImmutableAndClear();
```

Comment:
>return result.ToImmutableAndClear(); [](http://example.com/codeflow?start=8&length=36)

Old code had a RemoveDuplicates call here. Just looking for validation that it wasn't necessary.

Reply:
it wasn't necessary.  this is for getting siblings, which alreadywill only be for unique projects.